src/lib/storage/snapshots_finish.rb

```diff
@@ -0,0 +1,48 @@
+require "yast2/fs_snapshot"
+require "installation/finish_client"
+
+module Storage
+  class SnapshotsFinish < ::Installation::FinishClient
+    include Yast::I18n
+
+    def initialize
+      textdomain "storage"
+
+      Yast.import "Mode"
+      Yast.import "StorageSnapper"
+      Yast.include self, "installation/misc.rb"
+    end
+
+    # Write configuration
+    #
+    # It creates a snapshot when no second stage is required and
+    # Snapper is configured.
+    #
+    # @return [TrueClass,FalseClass] True if snapshot was created;
+    #                                otherwise it returns false.
+    def write
+      if !second_stage_required? && StorageSnapper.configure_snapper?
```

Comment:
StorageSnapper.configure_snapper? does not work during upgrade since it looks at the userdata for the root volume. The required data therein is set by the storage proposal and expert partitioner but so far not by the update code.